Firestore/core/src/firebase/firestore/model/field_path.cc

```diff
@@ -30,6 +31,8 @@ namespace model {
 
 namespace {
 
+using api::ThrowInvalidArgument;
```

Comment:
I'm pretty surprised that it works to put this using statement inside the anonymous namespace and then use it outside the anonymous namespace. It's clearer to put this at the scope in which you're using it (i.e. as a child of model).

Reply:
Huh, that's a good point.  I can't really explain that (I verified that commenting it out does cause errors so we're not accidentally pulling in a using statement at the root or anything). :-/  Anyway, good catch.  Moved.

Reply:
It is supposed to work and actually [recommended](http://go/totw/119#unnamed-namespaces):
> A using-declaration placed in an unnamed namespace can be accessed from the enclosing namespace and vice versa. If you already have an unnamed namespace at the top of the file, prefer putting all aliases there. [...]

(not that I suggest moving them back, it's probably too much hassle)

Basically, everything inside the anonymous namespace is supposed to be accessible from the enclosing namespace within the source file.

Reply:
Huh, interesting!  Thanks for the info.  I'm going to leave it as-is (out of laziness and because it seems like a strange recommendation), but it's good to know.